speclang_modern.tex

```diff
@@ -2885,6 +2885,15 @@ \section{Abrupt termination}
 contracts to give behavioral properties to the \main{} function
 or to any function that may exit the program,
 \emph{e.g.} by calling the \exit{} function.
+The simple contract
+\begin{listing-nonumber}
+/*@ exits E;
+  @*/
+\end{listing-nonumber}
+means that the corresponding function (or statement) exits
```

Comment:
Shouldn't it be something like "if the corresponding function (or statement) calls `exit`, `E` will hold at the point were `exit` is called"?

Reply:
Your proposition is more precise, perhaps too much. The `exit` function is an example of function having `//@ exits true; ensures false;` as possible contract. But it is not the only one. The `abort` function behaves also like that. 

What about this:  _"the corresponding function (or statement) terminates throw an `exit` exception in a post-state where the property `E` holds."_ ?

Also, I like to emphase that this point (where the property holds) is also considered as a post-state (i.e. for the eventual `\assigns clause`).  In case your aborting function assigns some variables, the assignment of these variables has to be considered into the called functions.

Reply:
I think there is a misunderstanding here: my point was to insist on the fact that the presence of an `exits \true;` clause does not imply that the program will necessarily abort (through a call to exit or any other suitable mean), like an `ensures \true;` does not imply that the function returns normally.

Reply:
I agree with you.
Also, since there is no notion of `exception` into the C standard, it is better to use the notion of _program termination_ instead. What about something like _"If the program terminates while executing the corresponding function (or statement), then it exits in a post-state where  the property E holds."_ ?


note: this initial proposition was derived from the current text used for the `ensures` clause:
Into the document, something like that is written: _"The called function returns a state where the property E holds."_ As you said, that does not imply that the function will necessarily return. 
Why not adding a footnote there (in section 2.3.2)  ?

Reply:
> What about something like "If the program terminates while executing the corresponding function (or statement), then it exits in a post-state where the property E holds." ?

Sounds good to me

> Why not adding a footnote there (in section 2.3.2) ?

It's probably a good thing too